sources/core/Xenko.Core.Mathematics/Matrix.cs

```diff
@@ -3477,5 +3478,41 @@ public override bool Equals(object value)
             };
         }
 #endif
+
+        /// <summary>
+        /// Performs an implicit conversion from <see cref="Xenko.Core.Mathematics.Matrix"/> to <see cref="BulletSharp.Math.Matrix"/>.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>The result of the conversion.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static implicit operator BulletSharp.Math.Matrix(Matrix value)
+        {
+            unsafe { return *(BulletSharp.Math.Matrix*)&value; }
```

Comment:
Looking at https://github.com/AndresTraks/BulletSharpPInvoke/blob/master/BulletSharp/Math/Matrix.cs#L58
it seems we have a different memory layout compared to Bullet matrix (M12 vs M21)